From 8786b91488dae3f6dfeadd686e80d2ffc5c29320 Mon Sep 17 00:00:00 2001
From: Jackie Huang <jackie.huang@windriver.com>
Date: Thu, 25 Aug 2016 15:22:57 +0800
Subject: [PATCH] squid: don't do squid-conf-tests at build time

* squid-conf-tests is a test to run "squid -k parse -f"
  to perse the config files, which should not be run
  at build time since we are cross compiling, so remove
  it but it will be added back for the runtime ptest.

* Fix the directories of the conf files for squid-conf-tests
  so that it can run on the target board.

Upstream-Status: Inappropriate [cross compile specific]

Signed-off-by: Jackie Huang <jackie.huang@windriver.com>

---
 test-suite/Makefile.am | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/test-suite/Makefile.am b/test-suite/Makefile.am
index 0233c0e..e0021b6 100644
--- a/test-suite/Makefile.am
+++ b/test-suite/Makefile.am
@@ -40,8 +40,7 @@ TESTS += debug \
 	splay\
 	mem_node_test\
 	mem_hdr_test\
-	$(ESI_TESTS) \
-	squid-conf-tests
+	$(ESI_TESTS)
 
 ## Sort by alpha - any build failures are significant.
 check_PROGRAMS += debug \
@@ -159,19 +158,19 @@ VirtualDeleteOperator_SOURCES = \
 ##$(TARGLIB): $(LIBOBJS)
 ##	$(AR_R) $(TARGLIB) $(LIBOBJS)
 
-squid-conf-tests: $(top_builddir)/src/squid.conf.default $(srcdir)/squidconf/*
+squid-conf-tests: $(sysconfdir)/squid.conf.default squidconf/*
 	@failed=0; cfglist="$?"; rm -f $@ || $(TRUE); \
 	for cfg in $$cfglist ; do \
-		$(top_builddir)/src/squid -k parse -f $$cfg || \
+		squid -k parse -f $$cfg || \
 			{ echo "FAIL: squid.conf test: $$cfg" | \
-				sed s%$(top_builddir)/src/%% | \
-				sed s%$(srcdir)/squidconf/%% ; \
+				sed s%$(sysconfdir)/%% | \
+				sed s%squidconf/%% ; \
 				failed=1; break; \
 			}; \
 		if test "$$failed" -eq 0; then \
 			echo "PASS: squid.conf test: $$cfg" | \
-				sed s%$(top_builddir)/src/%% | \
-				sed s%$(srcdir)/squidconf/%% ; \
+				sed s%$(sysconfdir)/%% | \
+				sed s%squidconf/%% ; \
 		else break; fi; \
 	done; \
 	if test "$$failed" -eq 0; then cp $(TRUE) $@ ; fi
